### PR TITLE
Contact details

### PR DESCRIPTION
**Pull request by corsacca**
@ChrisChasm These changes are needed in order to modify or set the contact details from the rest api. 
The type of a Primary Phone number will always be "primary". But each channel (Phone, Email...) has a label that, which can be the translation. Same with the types (primary, moblie...)
THis lets the db key (contact_phone_primary_*) always be the same and allows the rest api to work whatever the language.

### CODE REVIEW

**Comment by ChrisChasm** [comment]
Great work bro! Much cleaner in general.

File: includes/models/class-contact-post-type.php [hunk 1]
```diff
@@ -800,44 +804,85 @@ public function add_new_contact_field () {
...
+	 * Get a list of the contact channels and their types
+	 * @access public
+	 * @since  0.1
+	 * @return mixed
+	 */
+    public function get_channels_list(){
+        $channelList = array(
+        	"phone" => array(
+		        "label" => __('Phone', 'disciple_tools'),
+		        "types" => array(
```
Comment (ChrisChasm) [inline includes/models/class-contact-post-type.php]: This is much nicer than what I had. Well done bro.

File: includes/models/class-contact-post-type.php [hunk 2]
```diff
@@ -595,23 +595,25 @@ public function get_custom_fields_settings () {
...
-                $tag = null;
+            foreach ($channels as $channel_key => $channel) {
+            	foreach($channel["types"] as $type_key => $type){
 
-                $key =  'contact_' . $channel . '_111' ;
-                $names = explode('_', $key);
+	                $tag = null;
 
-                if($names[1] != $names[2]) { $tag = ' ('. $names[2] . ')'; }
+	                $key =  'contact_' . $channel_key . '_' . $type_key . '_111' ;
```
Comment (ChrisChasm) [inline includes/models/class-contact-post-type.php]: Ya. This looks really good. Much cleaner logic.

File: includes/models/class-contact-post-type.php [hunk 3]
```diff
@@ -800,44 +804,85 @@ public function add_new_contact_field () {
...
+	/**
+	 * Helper function to create the unique metakey for contacts channels.
+	 * @param $channel_key
+	 * @param $channel_type
+	 * @param $field_type
+	 *
+	 * @return string
+	 */
+    public function create_channel_metakey ($channel_key, $channel_type, $field_type) {
+        return $field_type . '_' . $channel_key .'_'. $channel_type . '_' . $this->unique_hash(); // build key
```
Comment (ChrisChasm) [inline includes/models/class-contact-post-type.php]: Great. More clearly modular.

**Comment by corsacca** [comment]
@ChrisChasm  Thanks :)

**Comment by ChrisChasm** [comment]
Yikes. Found an error on the save new address meta box.

**Comment by ChrisChasm** [comment]
I'm getting...
Notice: Undefined index: contact_phone_mobile_419 in /Users/chrisw/Documents/PROJECTS/disciple-tools/wp-content/plugins/disciple-tools/includes/activity/hooks/class-hook-posts.php on line 165

Notice: Undefined index: contact_phone_mobile_419 in /Users/chrisw/Documents/PROJECTS/disciple-tools/wp-content/plugins/disciple-tools/includes/activity/hooks/class-hook-posts.php on line 174

Warning: Cannot modify header information - headers already sent by (output started at /Users/chrisw/Documents/PROJECTS/disciple-tools/wp-content/plugins/disciple-tools/includes/activity/hooks/class-hook-posts.php:165) in /Users/chrisw/Documents/PROJECTS/disciple-tools/wp-includes/pluggable.php on line 1195

When trying to add and a new channel to the contact

**Pull request merged**